wait/host.go

```diff
@@ -147,7 +146,10 @@ func getUaSpecSyncIndex(mur *toolchainv1alpha1.MasterUserRecord, targetCluster s
 
 func containsUserAccountStatus(uaStatuses []toolchainv1alpha1.UserAccountStatusEmbedded, uaStatus toolchainv1alpha1.UserAccountStatusEmbedded) bool {
 	for _, status := range uaStatuses {
-		if reflect.DeepEqual(uaStatus, status) {
+		if uaStatus.TargetCluster == status.TargetCluster &&
+			uaStatus.SyncIndex == status.SyncIndex &&
+			test.ConditionsMatch(uaStatus.Conditions, status.Conditions...) &&
+			test.ConditionsMatch(uaStatus.UserAccountStatus.Conditions, status.UserAccountStatus.Conditions...) {
```

Comment:
Aren't these two conditions matches the same? Check the type: https://github.com/codeready-toolchain/api/blob/master/pkg/apis/toolchain/v1alpha1/masteruserrecord_types.go#L76-L87 - it uses the composition

Reply:
ok, thanks for pointing this to me. I removed the duplicate assertion in 81e1665